From 58bb55d5cd9328e6aa9365bdc2982d39fced3fd4 Mon Sep 17 00:00:00 2001
From: Cyprian Lech <clech@plume.com>
Date: Mon, 24 Feb 2020 13:12:42 +0000
Subject: [PATCH 5/7] Add vlan support

vlan_id is prescribed from Wifi_Inet_Config to Wifi_Inet_State
---
 src/nm2/inc/nm2_iface.h       | 2 ++
 src/nm2/src/nm2_inet_config.c | 2 ++
 src/nm2/src/nm2_inet_state.c  | 3 ++-
 3 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/nm2/inc/nm2_iface.h b/src/nm2/inc/nm2_iface.h
index bc4286a..f689f74 100644
--- a/src/nm2/inc/nm2_iface.h
+++ b/src/nm2/inc/nm2_iface.h
@@ -153,6 +153,8 @@ struct nm2_iface
         uint8_t                     gre_remote_inet_addr[NM2_IFACE_INET_CONFIG_SZ(gre_remote_inet_addr)];
         bool                        gre_local_inet_addr_exists;
         uint8_t                     gre_local_inet_addr[NM2_IFACE_INET_CONFIG_SZ(gre_local_inet_addr)];
+        bool                        vlan_id_exists;
+        int                         vlan_id;
     }
     if_cache;
 };
diff --git a/src/nm2/src/nm2_inet_config.c b/src/nm2/src/nm2_inet_config.c
index 94109a1..a164e46 100644
--- a/src/nm2/src/nm2_inet_config.c
+++ b/src/nm2/src/nm2_inet_config.c
@@ -810,6 +810,8 @@ void nm2_inet_copy(
     NM2_IFACE_INET_CONFIG_COPY(piface->if_cache.gre_remote_inet_addr, iconf->gre_remote_inet_addr);
     piface->if_cache.gre_local_inet_addr_exists = iconf->gre_local_inet_addr_exists;
     NM2_IFACE_INET_CONFIG_COPY(piface->if_cache.gre_local_inet_addr, iconf->gre_local_inet_addr);
+    piface->if_cache.vlan_id_exists = iconf->vlan_id_exists;
+    NM2_IFACE_INET_CONFIG_COPY(piface->if_cache.vlan_id, iconf->vlan_id);
 }
 
 
diff --git a/src/nm2/src/nm2_inet_state.c b/src/nm2/src/nm2_inet_state.c
index 95ab308..53434b9 100644
--- a/src/nm2/src/nm2_inet_state.c
+++ b/src/nm2/src/nm2_inet_state.c
@@ -215,9 +215,10 @@ void nm2_inet_state_to_schema(
     NM2_IFACE_INET_CONFIG_COPY(pstate->gre_remote_inet_addr, piface->if_cache.gre_remote_inet_addr);
     pstate->gre_local_inet_addr_exists = piface->if_cache.gre_local_inet_addr_exists;
     NM2_IFACE_INET_CONFIG_COPY(pstate->gre_local_inet_addr, piface->if_cache.gre_local_inet_addr);
+    NM2_IFACE_INET_CONFIG_COPY(pstate->vlan_id, piface->if_cache.vlan_id);
+    pstate->vlan_id_exists = piface->if_cache.vlan_id_exists;
 
     /* Unsupported fields, for now */
-    pstate->vlan_id_exists = false;
     pstate->parent_ifname_exists = false;
 
     pstate->softwds_mac_addr_exists = false;
-- 
2.25.1

